IdnoPlugins/OAuth2/Application.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+namespace IdnoPlugins\OAuth2 {
+
+
+    class Application extends \Idno\Common\Entity
+    {
+
+        /**
+         * Generate a new keypair
+         */
+        public function generateKeypair()
+        {
+            $this->key = hash('sha256', mt_rand() . microtime(true) . $this->getTitle());
+            $this->secret = hash('sha256', mt_rand() . microtime(true) . $this->key);
+        }
+
+        /**
+         * Helper function to create a new application with a new keypair.
+         * @param type $title
+         * @return \IdnoPlugins\OAuth2\Application
+         */
+        public static function newApplication($title)
+        {
+            $app = new Application();
+            $app->setTitle($title);
+            $app->generateKeypair();
+
+            return $app;
+        }
+
+        /**
+         * Saves changes to this object based on user input
+         * @return true|false
+         */
+        function saveDataFromInput()
+        {
+
+            if (empty($this->_id)) {
+                $new = true;
+            } else {
+                $new = false;
+            }
+
+            $this->setTitle(\Idno\Core\site()->currentPage()->getInput('name'));
+
+            $this->setAccess('PUBLIC');
+            return $this->save();
+        }
+
+    }
+
+}
```

Comment:
Can we have linebreaks plz?

Reply:
PHPCBF added a line break, but that's not the same thing as a blank line. If this isn't the case we'd need to alter the rules.

Reply:
Meanwhile adding manually